app/scripts/etl-rest-services/etl-rest-services-settings.service.js

```diff
@@ -1,31 +1,66 @@
 /*jshint -W003, -W026, -W117, -W098 */
-(function() {
+(function () {
   'use strict';
 
   angular
-        .module('app.etlRestServices')
-        .service('EtlRestServicesSettings', EtlRestServicesSettings);
+    .module('app.etlRestServices')
+    .service('EtlRestServicesSettings', EtlRestServicesSettings);
 
-  EtlRestServicesSettings.$inject = [];
+  EtlRestServicesSettings.$inject = ['$cookies'];
 
-  function EtlRestServicesSettings() {
+  function EtlRestServicesSettings($cookies) {
     var serviceDefinition;
     var restUrlBaseList = ['https://etl.ampath.or.ke:8002/etl/', 'https://test1.ampath.or.ke:8002/etl/'];
     var restUrlBase = restUrlBaseList[1];
 
+    initialize();
     serviceDefinition = {
-          getCurrentRestUrlBase: getCurrentRestUrlBase,
-          setCurrentRestUrlBase: setCurrentRestUrlBase,
-          restUrlBase: restUrlBase
-        };
+      reInitialize: initialize,
+      getCurrentRestUrlBase: getCurrentRestUrlBase,
+      setCurrentRestUrlBase: setCurrentRestUrlBase,
+      restUrlBase: restUrlBase,
+      addUrlToList: addUrlToList,
+      getUrlBaseList: getUrlBaseList,
+      hasCoockiePersistedCurrentUrlBase: hasCoockiePersistedCurrentUrlBase
+    };
     return serviceDefinition;
 
+    function initialize() {
+
+      var lastSetUrl = $cookies.get('restEtlUrlBase');
+
+      if (lastSetUrl) {
+        restUrlBase = lastSetUrl;
+      }
+    }
+
+    function hasCoockiePersistedCurrentUrlBase() {
+      var lastSetUrl = $cookies.get('restEtlUrlBase');
+
+      if (lastSetUrl) {
+        return true;
+      }
+
+      return false;
+    }
+
     function getCurrentRestUrlBase() {
-          return restUrlBase;
-        }
+      return restUrlBase;
+    }
 
     function setCurrentRestUrlBase(url) {
-          restUrlBase = url;
-        }
+      restUrlBase = url;
+      var d = new Date();
+      d.setFullYear(2050);//expires in 2050
+      $cookies.put('restEtlUrlBase', url, { 'expires': d });
+    }
+
+    function getUrlBaseList() {
+      return restUrlBaseList;
+    }
+
+    function addUrlToList(url) {
+      restUrlBaseList.push(url);
```

Comment:
So this lives only for the duration of the application being open in the browser, right?